### DISCUSSION TITLE
Adding interpolation from file to PiecewiseLinearInterpolationMaterial

### DISCUSSION THREAD

**Discussion by AndrewFalkowski** [idaholab/moose#22181] [Q&A General] [answered]
How difficult would it be to add support for pulling x,y values from a file to the PiecewiseLinearInterpolationMaterial object? This seems like a fairly simple and worthwhile addition given how common pulling materials data from files is. From my understanding the only way to currently replicate this is to use a PiecewiseLinear function and then pull that with a CoupledValueFunctionMaterial, which is somewhat cumbersome.

**Comment by GiudGiud** [comment, marked as answer]
Should be GenericFunctionMaterial for the current way to do this*

This would not be too hard. It's just x,y values, no time dependence?

You could do it through the PropertyReadFile user object or you could duplicate the machinery in the function
